meta2v2/meta2_filters_action_container.c

```diff
@@ -309,11 +309,11 @@ _load_list_params(struct list_params_s *lp, struct gridd_filter_ctx_s *ctx,
 	const char *fstr = meta2_filter_ctx_get_param(ctx, NAME_MSGKEY_FLAGS);
 	if (NULL != fstr) {
 		guint32 flags = atoi(fstr);
+		lp->flags = flags;
```

Comment:
Instead of merging all booleans in an obscure "flags" field, I vote for generalizing the thread-local structure usage. This means calling `meta2_filter_extract_force_master` in all meta2 request handlers doing read-only operations.